### PR TITLE
feat(@spartacus/storefront): refactor banner and responsive banner component logic

### LINKED ISSUE #1050: refactor banner and responsive banner component logic
The components should delegate logic to a component service

### PR DESCRIPTION
**Pull request by Pucek9**
Closes #1050

### CODE REVIEW

File: projects/storefrontlib/src/lib/cms-lib/banner/responsive-banner.component.ts [hunk 1]
```diff
@@ -11,42 +9,7 @@ import { Observable } from 'rxjs';
...
-            return (srcset += `${this.getBaseUrl()}${
-              data.media[format.code].url
-            } ${format.width}w, `);
-          } else {
-            return srcset;
-          }
-        }, '');
-      })
-    );
+    return 'responsive-banner ' + this.service.getComponentUID();
```
Comment (Matejk00) [inline projects/storefrontlib/src/lib/cms-lib/banner/responsive-banner.component.ts]: Here we can use string templating instead of concatenation.

Comment (Matejk00) [inline projects/storefrontlib/src/lib/cms-lib/banner/responsive-banner.component.ts]: Also can we assign 'responsive-banner' to some `const RESPONSIVE_BANNER_CLASS` or something like this to describe little bit what that string means. In addition in a future we can use that variable into some other methods if needed.

File: projects/storefrontlib/src/lib/cms-lib/banner/banner.component.service.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  constructor(
+    public component: CmsComponentData<CmsBannerComponent>,
+    protected config: CmsConfig
+  ) {}
+
+  private convertToAbsoluteUrl = map(url => this.getBaseUrl() + url);
+
+  // TODO: move to a more generic location
+  // TODO: Make configurable
+  private formats: any[] = [
```
Comment (Matejk00) [inline projects/storefrontlib/src/lib/cms-lib/banner/banner.component.service.ts]: Can we provide an interface here? This method should return an array of formats, not `any` imho.

File: projects/storefrontlib/src/lib/cms-lib/banner/responsive-banner.component.ts [hunk 3]
```diff
@@ -11,42 +9,7 @@ import { Observable } from 'rxjs';
...
-  // TODO: move to a more generic location
-  // TODO: Make configurable
-  private formats: any[] = [
-    { code: 'mobile', width: 200 },
-    { code: 'tablet', width: 500 },
-    { code: 'desktop', width: 800 },
-    { code: 'widescreen', width: 1200 }
-  ];
-
   getClass(): string {
```
[the remark this answers is not in the document]
Comment (Pucek9) [reply projects/storefrontlib/src/lib/cms-lib/banner/responsive-banner.component.ts]: Methods for internal component usage (css) remain here.

File: projects/storefrontlib/src/lib/cms-lib/banner/banner.component.service.ts (hunk 2 above)
Comment (Pucek9) [reply projects/storefrontlib/src/lib/cms-lib/banner/banner.component.service.ts]: Is current version ok?

Comment (Pucek9) [reply projects/storefrontlib/src/lib/cms-lib/banner/banner.component.service.ts]: updated

**Pull request merged**